cfgmgr/intfmgr.cpp

```diff
@@ -969,8 +978,55 @@ bool IntfMgr::doIntfGeneralTask(const vector<string>& keys,
         }
         else
         {
-            FieldValueTuple fvTuple("mac_addr", MacAddress().to_string());
-            data.push_back(fvTuple);
+            if (!sag.empty())
+            {
+                // only VLAN interface can set static anycast gateway
+                if (!alias.compare(0, strlen(VLAN_PREFIX), VLAN_PREFIX))
+                {
+                    vector<FieldValueTuple> fvs;
+                    string gwmac = "";
+                    if (m_cfgSagTable.get("GLOBAL", fvs))
+                    {
+                        for (auto idx: fvs)
+                        {
+                            const auto &field = fvField(idx);
+                            const auto &value = fvValue(idx);
+
+                            if (field == "gateway_mac")
+                            {
+                                gwmac = value;
+                            }
+                        }
+                    }
+
+                    if (!gwmac.empty())
+                    {
+                        if (sag == "true")
+                        {
+                            updateSubIntfAdminStatus(alias, "down");
```

Comment:
This prevents unwanted events received when changing the interface MAC address in kernel.
Will add comment for it.

Reply:
Added comment for it.

Reply:
Changed to call new function to bring device down and up. 
It's because IPv6 LL is generated from the device's MAC address, it needs to be regenerated when it's MAC address is changed.